### PR TITLE
 Use upstream thumbnail if available

### LINKED ISSUE #675: Use `thumbnail_url` for thumbnail generation when present
## Problem

Some providers (like SMK) may link to quite large images in their `image_url`. This can cause unacceptably slow response times or even timeouts when generating thumbnails via our thumbnail service.

## Description

We already have a `thumbnail_url` available on the Image model which we can use.

When the `thumbnail_url` is available, we should send this to the thumbnail service instead of the `image_url`. This should be a small change [here](https://href.li/?https://github.com/WordPress/openverse-api/blob/f0c44d571249a2631eef5a13ea0d56702210daf3/api/catalog/api/views/image_views.py#L109):

```
image_url = image.thumbnail or image.url
```
_(using the serialized field names)_

## Additional context

https://github.com/WordPress/openverse-catalog/issues/698 tracks updating the SMK provider script in the Catalog to populate `thumbnail_url` with a link to a smaller image size.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #675 by @stacimc

## Description

Sends the URL of the upstream thumbnail to Photon if it's available and it's **not** from Phylopic (since these are not working currently), so we can get the thumbnail faster and prevent timing out when the image is too big. See [this comment](https://github.com/WordPress/openverse/issues/675#issuecomment-1464969665) from the linked issue for more details on the thumbnail investigation.

## Testing Instructions

You can manually set a thumbnail URL from the database for any image and test if it's retrieved from the `/thumb` endpoint.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]: https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/api/views/image_views.py [hunk 1]
```diff
@@ -107,17 +106,10 @@ def oembed(self, request, *_, **__):
...
-        if not image_url:
-            raise NotFound("Could not find image.", 404)
-
-        # Hotfix to use scaled down version of the image from SMK
-        # TODO Remove when this issue is addressed:
-        # TODO https://github.com/WordPress/openverse-catalog/issues/698
-        if "iip.smk.dk" in image_url:
-            width = settings.THUMBNAIL_WIDTH_PX
-            image_url = re.sub(r"!\d+,", f"!{width},", image_url)
+        # TODO: Remove the Phylopic check once the thumbnails are fixed.
```
[the remark this answers is not in the document]
Comment (krysal) [reply api/catalog/api/views/image_views.py]: Forgot to remove the comment, good catch! My plan is to leave it as is and correct the underlying data (in the catalog DB), so yes, it's the second option you mention.

File: api/catalog/api/views/image_views.py [hunk 2]
```diff
@@ -107,17 +106,7 @@ def oembed(self, request, *_, **__):
     )
     def thumbnail(self, request, *_, **__):
         image = self.get_object()
-
-        image_url = image.url
-        if not image_url:
-            raise NotFound("Could not find image.", 404)
```
Comment (sarayourfriend) [inline api/catalog/api/views/image_views.py:113]: Is removing this check intentional/are we just relying on the upstream response from photon to return 404 in this case?

**Review by sarayourfriend — APPROVED** [review]
LGTM, just one question but not a blocker as I don't think it'll break anything.

File: api/justfile [hunk 3]
```diff
@@ -83,7 +83,7 @@ collectstatic:
 #########
 
 # Run API tests inside the Docker container
-test *args: _all-up
+test *args:
```
Comment (dhruvkb) [inline api/justfile:86]: I think `test` needs to depend on `wait-up`, hence the merge conflict. I think using out this file as it exists on the `main` branch should be fine.

**Review by dhruvkb — APPROVED** [review]
LGTM!

File: api/justfile (hunk 3 above)
Comment (krysal) [reply api/justfile]: Thanks Dhruv!

**Pull request merged**